lib/ts/recipe/accountlinking/recipe.ts

```diff
@@ -322,9 +347,21 @@ export default class Recipe extends RecipeModule {
         // we do not pass in third party info, or both email or phone
         // cause we want to guarantee that the output array contains just one
         // primary user.
+
+        // map according to the correct types here so we don't have to change the types upstream
+        const accountInfoInput: AccountInfoInput = {
+            ...accountInfo,
+            webauthn: undefined,
+        };
+        // any credentialId will identify the same user
+        if (accountInfo.webauthn?.credentialIds.length) {
+            accountInfoInput.webauthn = {
+                credentialId: accountInfo.webauthn.credentialIds[0],
+            };
+        }
```

Comment:
Here it may also not make sense to search using the credentialId, since it'll only match the current user which is already a part of the results because the email address is already present.

Reply:
Done